packages/e2e/test/web-extension/dapp-sdk/features/singleDelegation.ts

```diff
@@ -0,0 +1,62 @@
+import { Cardano } from '@cardano-sdk/core';
+import { ObservableWallet } from '@cardano-sdk/wallet';
+import { buildAndSignTx } from '../utils';
+import { combineLatest, take, tap } from 'rxjs';
+import { toSerializableObject } from '@cardano-sdk/util';
+
+export const singleDelegation = ({
+  connectedWallet,
+  logger
+}: {
+  logger: typeof console;
+  connectedWallet: ObservableWallet;
+}) => {
+  const delegateElement = document.querySelector('#info-delegate-assets')!;
+  const delegateAssetsElement = document.querySelector('#info-delegate-asset-id')!;
+
+  combineLatest([
+    connectedWallet.addresses$,
+    connectedWallet.balance.utxo.available$,
+    connectedWallet.delegation.rewardAccounts$
+  ])
+    .pipe(
+      take(1),
+      tap(async ([addresses, availableBalance, rewardAccounts]) => {
+        if (!connectedWallet) {
+          return logger.warn('Please, connect the wallet first');
+        }
+
+        if (!availableBalance.assets || availableBalance.assets?.size === 0) {
+          throw new Error('Your wallet has no assets');
+        }
```

Comment:
It shouldn't matter whether wallet has any native assets for delegation. Please remove the assertion or update to check coins. Same for undelegation function.
```suggestion
        if (availableBalance.coins === 0n) {
          throw new Error('Your wallet has no coins');
        }
```